### DISCUSSION TITLE
Foreign key editing, on update/delete dropdown value is "Restrict" when "No Action" is the actual value

### DISCUSSION THREAD

**Discussion by julijan** [dbeaver/dbeaver#38626] [Q&A] [answered]
### Description

When editing a previously created and saved foreign key, "On Update" and "On Delete" are not properly selected if their value was "No Action". "Restrict" is selected instead.

### DBeaver Version

Community Edition 24.2.0.202409011551

### Operating System

Linux Mint 21.2, kernel 6.8.0-40-generic

### Database and driver

MySQL: 8.0.34-0ubuntu0.22.04.1 for Linux on x86_64 ((Ubuntu))
Driver: Driver for MySQL 8 and later

### Steps to reproduce

1. create two tables, for example orders and order_items
2. in order_items, create a foreign key referencing orders, select "No Action" for on delete and on update
3. save & close table's tab
4. open table, navigate to foreign keys, select foreign key created in step 2, and press F4
5. observe the dropdown for on delete/update - both will have "Restrict" selected, even though they were created as "No action" when created in step 2

### Additional context

While the values selected as "Restrict" are incorrect, the actual actions are properly stored when foreign key is created, so it looks like the only issue is that the app fails to properly select "No action" when editing a foreign key, I'm assuming it is looking for "ON DELETE" and "ON UPDATE" in the table schema, and if none is found it defaults to "Restrict", but I could be wrong.

**Comment by E1izabeth** [comment]
Thank you for bug report

**Comment by labanovichttps** [comment, marked as answer]
@julijan  Thank you for your report. But it's not the issue, when we read metadata from the db, we already get "Restrict", even if we set "NO ACTION" in manual SQL query. 

```
ALTER TABLE labanovich.`3a0_table`
  ADD CONSTRAINT `3a0_table_JOBS_FKKL`
  FOREIGN KEY (`JOB_ID`)
    REFERENCES labanovich.`JOBS`(`JOB_ID`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION;
``` 

see https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html#foreign-key-actions

<img width="1155" height="180" alt="Image" src="https://github.com/user-attachments/assets/2c371867-50a6-497b-964f-522fb94f4864" />

**Comment by julijan** [comment]
@labanovichttps thanks for clearing that up! I was not aware of this MySQL behavior. I guess it makes sense.

This is not a bug in DBeaver, but for InnoDB, it might be useful to do one of:
a) remove "No action" from the options, as it's effectively same as "Restrict"
b) add a warning that it will behave same as "Restrict" when "No action" is selected

I will close this issue, it's not a bug, at best, this becomes a minor feature suggestion.
